### ISSUE TITLE
gsi-openssh silently fails to install if not running as root

### ISSUE THREAD

**Issue by Phidica** [gridcf/gct#118]
I'm building the latest master on Debian because I wanted to get gsi-openssh working.

After running the following,
```
$ autoreconf --install --no-recursive
$ ./configure --prefix=$PWD/install --enable-gsi-openssh
$ make -j3
$ make -j3 install
```
I did not see `gsissh` in the ./install/bin directory. Note that I did not execute any of these commands with `sudo` because my install prefix is a local user directory and hence *it should not be necessary*.

I spotted the issue buried in the output of the `make install` run:
```
<trim>
./mkinstalldirs /home/phidica/Code/gct/install/share/man
./mkinstalldirs /home/phidica/Code/gct/install/share/man/man1
./mkinstalldirs /home/phidica/Code/gct/install/share/man/man5
./mkinstalldirs /home/phidica/Code/gct/install/share/man/man8
./mkinstalldirs /home/phidica/Code/gct/install/libexec
(umask 022 ; ./mkinstalldirs /var/empty)
mkdir /var/empty
mkdir: cannot create directory ‘/var/empty’: Permission denied
make[4]: *** [Makefile:346: install-files] Error 1
make[4]: Leaving directory '/home/phidica/Code/gct/gsi_openssh/source'
make[3]: *** [Makefile:2743: gsi-openssh-install] Error 2
<trim>
```

It appears that there is some call in the gsi-openssh install process that does not respect the install prefix I overrode with `./configure`, and the install fails because it attempts to create `/var/empty` as a normal user.

I solved the problem by running `sudo mkdir /var/empty` and repeating the install.

I have two independent interpretations of what the "issue" to be solved here is:
1. The gsi-openssh install process neglects to incorporate `$prefix` for this one directory, so it should instead attempt to create, eg, `$prefix/var/empty`.
1. The failure of this component was silent and no advice was provided on how to resolve the problem, eg, bring the install to a halt and print "Couldn't create /var/empty. Do you lack superuser permissions?"

I believe that implementing either one of these, individually, would resolve this issue.

**Comment by msalle** [comment]
Hi,
thank you for opening the issue and providing feedback.
Several remarks. First of all, this is actually an issue with [openssh-portable](https://github.com/openssh/openssh-portable/) itself, inherited by 'our gct' version, so perhaps we should discuss it there (see reporting bugs in their README).
Concerning the issue itself, note that `/var/empty` is a special directory used as a chroot as part of the privilege separation. See  [gsi_openssh/source/README.privsep](https://github.com/openssh/openssh-portable/blob/V_7_5_P1/README.privsep) for details.
Given this special role and its required permissions, I don't think it can or should be in a user-owned directory. In fact, installing an ssh server as non-root is in general not very useful nor secure.
Note that it is possible to set the directory explicitly using `--with-privsep-path=` as described in that same README.privsep. So something like

    ./configure --prefix=$PWD/install --with-privsep-path=$PWD/install/var/empty \
                --enable-gsi-openssh

actually works.

**Comment by Phidica** [comment]
Thanks for the response. I had assumed that setting `prefix` would be enough to take care of all those other sorts of paths, but I guess they have good reason not to. It's good to know there's a proper configuration method to get around this.

As such then I'd like this issue to turn to interpretation (2) I mentioned above, namely that I explicitly wanted gsi-openssh installed and the installation of that component silently failed. In my opinion the `make install` procedure should have treated the error as fatal and immediately halted the installation of gct. Is that an issue with openssh-portable, or with this project?

**Comment by msalle** [comment]
Hi,
I agree it would be nice if `make` would end with an non-zero exit value and that's indeed a minor bug. On the other hand, I don't think the error is that hidden in the output: it's pretty much the last that's printed.
The fix would be to add an `|| exit 1` to line [15 of the main Makefile.am](https://github.com/gridcf/gct/blob/master/Makefile.am#L15) and similarly to the other target lines. In fact the GT_ALL_STAMPS already has one.

**Comment by fscheiner** [comment]
Fixed in [GCT 6.2.20201212 maintenance release](https://github.com/gridcf/gct/releases/tag/v6.2.20201212).
